### PR TITLE
fix(suite-native): passphrase submit button is never hidden behind ke…

### LINKED ISSUE #12618: Passphrase UX revamp
Most critical UX issues on passphrase flow that should be addressed before release and would be confusing to the user and make the whole flow less fluent if not implemented. For example bottom sheets displayed in wrong screens, wrong redirects, useless steps in flow, etc.

UX Issues linked [here](https://www.figma.com/design/mmDnYtctfjnreWhTjLfeQH/Passphrase?node-id=2896-19290&t=R6WoQx6TcqxRCVpT-11)

- [x] Description to be done after design check by @shenkys 
- [x] Cancel error 🔴
- [x] Confirm empty scrolling 🔴 (fixed https://github.com/trezor/trezor-suite/pull/12737)
- [x] Missing "reveal" eye icon 🔴 (fixed by https://github.com/trezor/trezor-suite/pull/12727)

### PR DESCRIPTION
**Pull request by PeKne**
## Description
- passphrase form scroll down on appearance of submit button when needed. The button will be never hidden behind the keyboard anymore.

## Related Issue

Resolve #12618

### CODE REVIEW

File: suite-native/module-authorize-device/src/components/passphrase/PassphraseForm.tsx [hunk 1]
```diff
@@ -73,8 +81,29 @@ export const PassphraseForm = ({ inputLabel, onFocus }: PassphraseFormProps) =>
...
+                setTimeout(
+                    () => scrollView.scrollTo({ y: y - SCROLL_OFFSET, animated: true }),
+                    SCROLL_DELAY,
+                );
+            });
+        }
+    }, [scrollView]);
+
+    useEffect(() => {
+        if (watchPassphrase !== '') {
```
Comment (juriczech) [inline suite-native/module-authorize-device/src/components/passphrase/PassphraseForm.tsx]: `isDirty` should be used here to match the render condition for button in JSX.

File: suite-native/module-authorize-device/src/components/passphrase/PassphraseForm.tsx [hunk 2]
```diff
@@ -22,12 +23,17 @@ import {
...
     RootStackParamList,
     StackToStackCompositeNavigationProps,
+    useScrollView,
 } from '@suite-native/navigation';
 
 import { EnterPassphraseOnTrezorButton } from './EnterPassphraseOnTrezorButton';
 
 const FORM_CARD_PADDING = 12;
 
+// The scroll behaves differently for each platform, so we need to adjust the offset so make it always scroll to correct position.
```
Comment (juriczech) [inline suite-native/module-authorize-device/src/components/passphrase/PassphraseForm.tsx]: Please elaborate how the behavior is different that it needs different offset

**Review by juriczech — COMMENTED** [review]
Tested on Android emulator, works well! 💪

File: suite-native/module-authorize-device/src/components/passphrase/PassphraseForm.tsx (hunk 2 above)
Comment (PeKne) [reply suite-native/module-authorize-device/src/components/passphrase/PassphraseForm.tsx]: 38ecda525e

File: suite-native/module-authorize-device/src/components/passphrase/PassphraseForm.tsx (hunk 1 above)
Comment (PeKne) [reply suite-native/module-authorize-device/src/components/passphrase/PassphraseForm.tsx]: 38ecda525e

**Review by juriczech — APPROVED** [review]

**Pull request merged**